scripts/generate-app-config/generateExtensionConfig.js

```diff
@@ -0,0 +1,23 @@
+const ImportAggregator = require('./generation-objects/ImportAggregator');
+
+/**
+* Generates the file representing app name configuration.
+*/
+const generateExtensionConfig = (extensionConfig) => {
+  const imports = new ImportAggregator();
+
+
+  const config = extensionConfig.map(ext => (
```

Comment:
Do we need to do any validations to ensure `iconPath`, `key`, `text` and `componentPath`, and `size` are included before adding the configuration? At a minimum` iconPath` or `ComponentPath`?

Reply:
I can add documentation that all fields are required. We don't have much, if any helpful validation anywhere else for the config, if you don't supply things the site will fail. It hasn't been much of a problem so far, if people start asking more questions, it may be worth the effort in the future.